### PR TITLE
fix: Updating chart config would not update interactive filters

### PR DESCRIPTION
**Pull request by ptbrowne**
## Copy a chart and change chart config filters

Closes https://github.com/visualize-admin/visualization-tool/issues/162

When copying a chart and changing the filters in the chart config, it was not reflected in the interactive filters.

To fix this issue, changes to chart config filters are detected in the chart preview and used to update the interactive filters.

I find the solution a bit hacky still since

- the state is not correclty updated if the chart-preview component is not mounted
- It only work for single value filters right now (but the `UPDATE_DATA_FILTER` action is only suited for single value filters it seem)

Still, it follows the current approach that updates the interactive filters from the chart config when the chart preview is mounted.

It seems like the chart interactive filters should be at the same level as the chart config and be updated through a common "orchestrator". Maybe we could combine the two reducers into one for these usecases ? I think we'd have to discuss with @herrstucki for this kind of change that would be more invasive.

### CODE REVIEW

**Comment by ptbrowne** [comment]
> I wish the two fixes were separate PRs 😅

Yep good idea. I am splitting this into https://github.com/visualize-admin/visualization-tool/pull/169, and we'll keep the discussion on updating the interactive chart config here.

File: app/charts/shared/chart-helpers.tsx [hunk 1]
```diff
@@ -1,6 +1,9 @@
 import { sum } from "d3";
+import { omitBy } from "lodash";
+import React from "react";
```
[the remark this answers is not in the document]
Comment (ptbrowne) [reply app/charts/shared/chart-helpers.tsx:3]: Ah yep, must be the auto import from VSCode, I need to check my settings thank.

File: app/charts/shared/chart-helpers.tsx [hunk 2]
```diff
@@ -11,24 +14,44 @@ import {
 // Prepare filters used in data query:
 // merges "hard" data filters (in-editor, publisher-defined)
 // and "interactive" data filters (user-defined), if applicable
+// and removes none values if specified in the options
 export const useQueryFilters = ({
   chartConfig,
   interactiveFiltersIsActive,
+  filterNone,
```
[the remark this answers is not in the document]
Comment (ptbrowne) [reply app/charts/shared/chart-helpers.tsx]: This is dealt with in the #169 PR

File: app/configurator/use-filter-changes.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import { string } from "io-ts";
```
Comment (herrstucki) [inline app/configurator/use-filter-changes.ts:1]: Nit: This import doesn't seem used

**Review by herrstucki — COMMENTED** [review]
LGTM (apart from the unused import :))

File: app/configurator/use-filter-changes.ts (hunk 3 above)
Comment (ptbrowne) [reply app/configurator/use-filter-changes.ts]: Thanks, done

**Pull request merged**